source/extensions/quic_listeners/quiche/envoy_quic_client_connection.cc

```diff
@@ -92,6 +92,14 @@ void EnvoyQuicClientConnection::setUpConnectionSocket() {
   }
 }
 
+void EnvoyQuicClientConnection::switchConnectionSocket(
+    Network::ConnectionSocketPtr&& connection_socket) {
+  auto writer = new EnvoyQuicPacketWriter(*connection_socket);
```

Comment:
nit: prefer storing in unique_ptr and then releasing into the QUICHE code, in case intermediate code throws an exception.